components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImplIT.java

```diff
@@ -229,17 +232,177 @@ public void testGetApplicationPolicies() throws Exception {
     @Test(description = "Get Subscription Policies")
     public void testGetSubscriptionPolicies() throws Exception {
         SubscriptionPolicy policy = SampleTestObjectCreator.createDefaultSubscriptionPolicy();
+        SubscriptionPolicy policy2 = SampleTestObjectCreator.createSubscriptionPolicyWithBandwithLimit();
+        //policy 1 has following 2 true. checking for fault scenario from policy2
+        policy2.setStopOnQuotaReach(false);
+        policy2.setDeployed(false);
+        
         PolicyDAO policyDAO = DAOFactory.getPolicyDAO();
         //add policy
         policyDAO.addSubscriptionPolicy(policy);
+        policyDAO.addSubscriptionPolicy(policy2);
+        
         List<SubscriptionPolicy> policyList = policyDAO.getSubscriptionPolicies();
         Assert.assertNotNull(policyList);
-        Assert.assertNotNull(policyDAO.getSubscriptionPolicy(policy.getPolicyName()), "Retrieving Subscription policy "
-                + "by name failed for policy with name: " + policy.getPolicyName());
-        Assert.assertNotNull(policyDAO.getSubscriptionPolicyByUuid(policy.getUuid()), "Retrieving Subscription policy "
-                + "by id failed for policy with id: " + policy.getUuid());
+ 
+        SubscriptionPolicy retrievedPolicy = null;
+        SubscriptionPolicy retrievedPolicy2 = null;
+        //there are defaut policies already there
+        for (Iterator iterator = policyList.iterator(); iterator.hasNext();) {
+            SubscriptionPolicy subscriptionPolicy = (SubscriptionPolicy) iterator.next();
+            if (subscriptionPolicy.getPolicyName().equals(policy.getPolicyName())) {
+                retrievedPolicy = policy;
+            } 
+            
+            if (subscriptionPolicy.getPolicyName().equals(policy2.getPolicyName())) {
+                retrievedPolicy2 = policy2;
+            }
+        }
+        
+        Assert.assertNotNull(retrievedPolicy, "Policy " + policy.getPolicyName() + " not in DB");
+        Assert.assertNotNull(retrievedPolicy2, "Policy " + policy2.getPolicyName() + " not in DB");
+        
+        Assert.assertEquals(retrievedPolicy.getPolicyName(), policy.getPolicyName(),
```

Comment:
Shall we put the repeated assertion strings into a constant without duplicating in all these test cases? Might be useful if we are to change the message in future.

Reply:
ok looks like the message differs for each. Hence ignore this